daemon/api_asserts.go

```diff
@@ -88,10 +101,25 @@ func assertsFindMany(c *Command, r *http.Request, user *auth.UserState) Response
 	state.Unlock()
 
 	assertions, err := db.FindMany(assertType, headers)
-	if asserts.IsNotFound(err) {
-		return AssertResponse(nil, true)
-	} else if err != nil {
+	if err != nil && !asserts.IsNotFound(err) {
 		return InternalError("searching assertions failed: %v", err)
 	}
+
+	if jsonResult {
+		assertsJSON := make([]struct {
+			Headers map[string]interface{} `json:"headers,omitempty"`
+			Body    string                 `json:"body,omitempty"`
+		}, len(assertions))
+		for i := range assertions {
+			assertsJSON[i].Headers = assertions[i].Headers()
+			if !headersOnly {
+				assertsJSON[i].Body = string(assertions[i].Body())
+			}
+		}
+		return SyncResponse(map[string]interface{}{
+			"assertions": assertsJSON,
```

Comment:
why the extra level? Typically we'd just return `SyncResponse(assertsJSON, nil)` here.